core/src/test/scripts/system_tests_validate_studies.py

```diff
@@ -80,6 +113,86 @@ def test_exit_status_multiple_studies(self):
         exit_status = validateStudies.main(args)
         self.assertEquals(1, exit_status)
 
+    def test_logs_study_label_before_validation_messages(self):
+        """The log file should start with a line describing the study.
+
+        A subsequent study should have its own header line.
+        """
+        # given
+        with TemporaryDirectory() as out_dir_path:
+            args = [
+                '--root-directory', 'test_data',
+                '--list-of-studies', 'study_various_issues,study_es_0',
+                '--portal_info_dir', PORTAL_INFO_DIR,
+                '--html-folder', out_dir_path
+                ]
+            # when
+            with redirect_stdout(StringIO()):
+                parsed_args = validateStudies.interface(args)
+                validateStudies.main(parsed_args)
+            # then
+            log_file_path = glob.glob(os.path.join(out_dir_path, 'log*.txt'))[0]
+            with open(log_file_path) as log_file:
+                log_file_lines = log_file.readlines()
+            self.assertIn('study_various_issues', log_file_lines[0])
+            last_line_of_first_study = next(
+                index
+                for index, line
+                in enumerate(log_file_lines)
+                if 'Validation complete' in line)
+            self.assertIn(
+                'study_es_0',
+                log_file_lines[last_line_of_first_study + 1])
+
+
+class ValidateStudiesWithEagerlyFlushingCollapser(unittest.TestCase):
+    """Test validation with the collapser flushing due to buffer capacity.
+
+    When validating very large studies, it will flush partway through a study.
+    This can be simulated with a smaller study by lowering the buffer capacity.
+    """
+
+    def setUp(self):
+        """Make the collapsing log message handler flush more eagerly."""
+        class EagerFlusher(logging.handlers.MemoryHandler):
+            def __init__(self, *args, **kwargs):
+                """Set the buffer capacity to 3 regardless of args."""
+                # leave out any capacity argument from args and kwargs
+                args = args[1:]
+                kwargs = {k: v for k, v in kwargs.items() if k != 'capacity'}
+                # pass 3 as the capacity argument
+                super(EagerFlusher, self).__init__(3, *args, **kwargs)
+        class EagerFlushingCollapser(
+                cbioportal_common.CollapsingLogMessageHandler,
+                EagerFlusher):
+            """CollapsingLogMessageHandler with EagerFlusher overrides."""
+            pass
+        self.original_collapser = cbioportal_common.CollapsingLogMessageHandler
+        cbioportal_common.CollapsingLogMessageHandler = EagerFlusher
```

Comment:
I realised, this mock no longer works now that `validateData` runs in its own process. The test isn't technically testing what it says it's testing any more. However, I don't see a better way to fix this than by making the buffer size configurable by passing a flag to `validateData`.